src/TreeContainer.js

```diff
@@ -107,6 +109,43 @@ class TreeContainer extends Component {
         };
     }
 
+    getNestedIds() {
+        const {
+            _dashprivate_layout: layout
+        } = this.props;
+
+        const ids = [];
+        const queue = [layout];
+
+        while (queue.length) {
+            const elementLayout = queue.shift();
+
+            const props = elementLayout &&
+                elementLayout.props;
+
+            if (!props) {
+                continue;
+            }
+
+            const { children, id } = props;
+
+            if (id) {
+                ids.push(id);
+            }
+
+            if (children) {
+                const filteredChildren = filter(
+                    child => !isSimpleComponent(child) && !Registry.resolve(child.type, child.namespace)._dashprivate_deepstate,
+                    Array.isArray(children) ? children : [children]
+                );
+
+                queue.push(...filteredChildren);
+            }
+        }
```

Comment:
Loop in-place on children and append the newly found children until all elements have been tested